### DISCUSSION TITLE
how to enable end to end https for web apps with multiple domain names

### DISCUSSION THREAD

**Discussion by gouthamgs** [gravitational/teleport#11290] [Q&A] [answered]
I need to support multiple web apps with different domain names like abc.com, xyz.com and def.com.
All these applications have https end points with certificates of respective domains installed. I need to provide access to these webapps through teleport proxy (as the web apps are behind the firewall) with end to end https with TLS termination happening at the web app itself. Looks like teleport proxy forces us to use the domain name of teleport proxy to access all the webapps, which is not useful in my case. 
Is there a way to configure teleport proxy as pass through and support multiple web apps with different domain names, with TLS termination happening at the web apps and thus enabling end to end HTTPS for these webapps?

**Comment by webvictim** [comment]
There's currently no way to configure this. Teleport's internal application routing logic requires all applications to be subdomains of the Teleport proxy.

**Comment by gouthamgs** [comment]
@webvictim OK. Thanks for the clarification.

**Comment by wszychta** [comment]
Hello @webvictim - is this still true for teleport version 14.x? We have multiple domains and we are planning to use single teleport cluster. We have usecase where we would like to use public_addr for apps with diffrent domain than teleport. Let's say teleport runs at teleport.example.com and we would like to run our app with public_addr set to app.example2.com. Is this possible with latest versions of teleport?

**Reply by webvictim** [reply, marked as answer]
Yes, this is possible now. You can set the `public_addr` for apps to a different domain. The requirements are:
- the DNS record for that domain must point to the Teleport proxy service
- Teleport must be able to make a secure TLS connection to that domain - i.e. it needs a valid, trusted certificate which cannot be self-signed
  - you can give the TLS certificate and private key to Teleport using the `https_keypairs` configuration on the Teleport proxy service, or use a reverse proxy/load balancer which terminates TLS with the correct certificate

**Reply by wszychta** [reply]
@webvictim @stevenGravy Unfortunatelly it doesn't work that way. I have created an issue with full description https://github.com/gravitational/teleport/issues/33138

**Reply by stevenGravy** [reply]
Can you confirm that the cert is showing in your browser for all domains?  I have set this up and it worked fine.

**Reply by wszychta** [reply]
I have separated certificates for both domains. Both of them are attached to teleport proxy.
